test/clj/chronograph/domain/timer_test.clj

```diff
@@ -0,0 +1,480 @@
+(ns chronograph.domain.timer-test
+  (:require [chronograph.domain.timer :as timer]
+            [clojure.test :refer :all]
+            [chronograph.fixtures :as fixtures]
+            [chronograph.factories :as factories]
+            [chronograph.domain.acl :as acl]
+            [chronograph.db.core :as db]
+            [clojure.spec.alpha :as s]
+            [chronograph.db.time-span :as db-time-span])
+  (:import org.postgresql.util.PSQLException))
+
+(def test-context
+  (atom {}))
+
+(defn- setup-members-org-tasks!
+  []
+  (let [{owner-id :users/id}
+        (factories/create-user)
+
+        {organization-id :organizations/id}
+        (factories/create-organization owner-id)
+
+        {member-id :users/id} (factories/create-user)
+        _ (acl/create! db/datasource
+                       {:user-id member-id
+                        :organization-id organization-id
+                        :role acl/member})
+        {task-id-1 :tasks/id} (factories/create-task organization-id)
+        {task-id-2 :tasks/id} (factories/create-task organization-id)]
+    (reset! test-context
+            {:owner-id owner-id
+             :organization-id organization-id
+             :member-id member-id
+             :task-id-1 task-id-1
+             :task-id-2 task-id-2})))
+
+(defn- setup-members-org-tasks-fixture
+  [f]
+  (setup-members-org-tasks!)
+  (f))
+
+(use-fixtures :once fixtures/config fixtures/datasource)
+(use-fixtures :each
+  fixtures/clear-db
+  setup-members-org-tasks-fixture)
+
+
+;; ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Create Timer Tests
+;; ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+
+(deftest create-timer-test
```

Comment:
Add a test for users creating a timer under an organization that they are not a part of.

Reply:
Fixed. To do this, I had to refactor the domain fn to accept organization ID. That in turn improved domain fn behaviour a bit.